File: 128-longest-consecutive-sequence/longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    // dump all in a set
    const set = new Set();
    for(let num of nums) {
        set.add(num);
    }

    // iterate in set
    // check if a possible start
    // keep on checking till you find the next one and update highest streak
    let highestStreak = -1;
    for(let num of set) {
        if(!set.has(num-1)) { // POSSIBLE START
            let n = num, streak = 1;
            while(set.has(n+1)) {
                streak++;
                n++;
            }
            highestStreak = Math.max(streak, highestStreak);
        }
    }

    return highestStreak == -1 ? 0 : highestStreak;
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> values(nums.begin(), nums.end());
        int longest = 0;

        for (int num : values) {
            if (values.find(num - 1) == values.end()) {
                int current = num;
                int length = 1;

                while (values.find(current + 1) != values.end()) {
                    ++current;
                    ++length;
                }

                longest = max(longest, length);
            }
        }

        return longest;
    }
};